libs/agno/agno/workflow/v2/parallel.py

```diff
@@ -30,40 +38,441 @@ class Parallel:
     name: Optional[str] = None
     description: Optional[str] = None
 
+    def __init__(
+        self,
+        *steps: Union[Step, "Steps", "Loop", "Parallel", "Condition"],
+        name: Optional[str] = None,
+        description: Optional[str] = None,
+    ):
+        self.steps = list(steps)
+        self.name = name
+        self.description = description
+
+    def _aggregate_results(self, step_outputs: List[StepOutput]) -> StepOutput:
+        """Aggregate multiple step outputs into a single StepOutput"""
+        if not step_outputs:
+            return StepOutput(step_name=self.name or "Parallel", content="No parallel steps executed")
+
+        if len(step_outputs) == 1:
+            # Single result, just update the step name
+            single_result = step_outputs[0]
+            single_result.step_name = self.name or "Parallel"
+            return single_result
+
+        # Multiple results - aggregate them
+        aggregated_content = self._build_aggregated_content(step_outputs)
+
+        # Combine all media from parallel steps
+        all_images = []
+        all_videos = []
+        all_audio = []
+        has_any_failure = False
+
+        for result in step_outputs:
+            all_images.extend(result.images or [])
+            all_videos.extend(result.videos or [])
+            all_audio.extend(result.audio or [])
+            if result.success is False:
+                has_any_failure = True
+
+        return StepOutput(
+            step_name=self.name or "Parallel",
+            content=aggregated_content,
+            images=all_images if all_images else None,
+            videos=all_videos if all_videos else None,
+            audio=all_audio if all_audio else None,
+            success=not has_any_failure,
+        )
+
+    def _build_aggregated_content(self, step_outputs: List[StepOutput]) -> str:
+        """Build aggregated content from multiple step outputs"""
+        aggregated = "## Parallel Execution Results\n\n"
+
+        for i, output in enumerate(step_outputs):
+            step_name = output.step_name or f"Step {i + 1}"
+            content = output.content or ""
+
+            # Add status indicator
+            if output.success is False:
+                status_icon = "❌"
+            else:
+                status_icon = "✅"
+
+            aggregated += f"### {status_icon} {step_name}\n"
```

Comment:
Maybe also add `SUCCESS` and `FAILURE` or similar

Reply:
done
<img width="654" alt="image" src="https://github.com/user-attachments/assets/08e852f6-090d-4cec-b094-bc702222bcc0" />